pkg/asset/fsck.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package asset
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+)
+
+// detectFileSystemType uses blkid to identify the file system type of a given image file
+func detectFileSystemType(imagePath string) (string, error) {
+	cmd := exec.Command("blkid", "-o", "value", "-s", "TYPE", imagePath)
+	output, err := cmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("failed to detect file system type: %v", err)
+	}
+	return string(output), nil
+}
+
+// runFsck uses the fs-specific fsck subcommand to check if an image file is clean
+func runFsck(imagePath string, fsType string) (bool, error) {
+	fsckCommand := fmt.Sprintf("fsck.%s", fsType)
+	cmd := exec.Command(fsckCommand, "-n", imagePath)
```

Comment:
I think we should also set some explicit timeout and sandbox the command via
https://github.com/google/syzkaller/blob/571351cb80e526bf973c8f11c46fa7f3bbc2da1e/pkg/osutil/osutil_linux.go#L53

We could theoretically expect all sorts of fsck behavior on the corrupted images.